src/Radios/Radios.jsx

```diff
@@ -40,7 +40,12 @@ const Radios = ({
   const name = idParts.join('.');
 
   const internalOnChange = ({ target }) => {
-    if (typeof onChange === 'function') {
+    let output = target.name;
+    const nameParts = target.name.split('.');
+    if(nameParts.length > 1){
+      output = nameParts[1];
```

Comment:
This can't just be `nameParts[1]` because it could easily be more deeply nested than this. This would work, however:
```suggestion
      output = nameParts.pop();
```

Reply:
Good suggestion, updated to pop off the last part now